s4e-web/src/app/views/map-view/sentinel-search/sentinel-form/sentinel-form.component.ts

```diff
@@ -0,0 +1,136 @@
+import {Component, forwardRef, Input, OnDestroy, OnInit} from '@angular/core';
+import {
+  isSentinelFloatParam, isSentinelSelectParam,
+  SentinelParam,
+  SentinelSelectParam
+} from '../../state/sentinel-search/sentinel-search.metadata.model';
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  FormControl,
+  FormGroup, NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator,
+  Validators
+} from '@angular/forms';
+import {disableEnableForm, stripNulls} from '../../../../utils/miscellaneous/miscellaneous';
+import {Subscription} from 'rxjs';
+
+
+@Component({
+  selector: 's4e-sentinel-form',
+  templateUrl: './sentinel-form.component.html',
+  styleUrls: ['./sentinel-form.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => SentinelFormComponent),
+      multi: true
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => SentinelFormComponent),
+      multi: true,
+    }
+  ]
+})
+export class SentinelFormComponent implements OnInit, OnDestroy, ControlValueAccessor, Validator {
+  form: FormGroup = new FormGroup({});
+  paramsDefPairs: ([SentinelParam, SentinelParam] | [SentinelParam])[];
+  private sub: Subscription|null = null;
+
+  @Input('paramsDef') set paramsDef(val: SentinelParam[]) {
```

Comment:
Could you change `val` name into something like `sentinelParams`?

Reply:
And cut this setter to smaller functions, so it will be know what is happen inside in few lines

Reply:
I'll decompose the setter (it's too long, I agree) but the `val` in setter does have a meaning - it's a setter, it has only one argument, and that's value. That way you know that it's a value passed to the setter and not some temporary variable on the stack - without looking at the method definition.

Though after shortening the method it won't matter as much.